content/src/ports/activeEntities.ts

```diff
@@ -0,0 +1,175 @@
+  inactivate(...pointers: Pointer[]): void
+  activate(pointer: Pointer, entity: Entity): void
+}
+
+const mapDeploymentsToEntities = (deployments: Deployment[]): Entity[] => {
+  return deployments.map(({ entityVersion, entityId, entityType, pointers, entityTimestamp, content, metadata }) => ({
+    version: entityVersion,
+    id: entityId,
+    type: entityType,
+    pointers,
+    timestamp: entityTimestamp,
+    content: content?.map(({ key, hash }) => ({ file: key, hash })),
+    metadata
+  }))
+}
+
+export const createActiveEntitiesComponent = (
+  components: Pick<AppComponents, 'database' | 'env' | 'logs' | 'metrics'>
+): ActiveEntities => {
+  const logger = components.logs.getLogger('ActiveEntities')
+  const cache = new LRU<EntityId, Entity>({
+    max: components.env.getConfig(EnvironmentConfig.ENTITIES_CACHE_SIZE)
+  })
+  const entityIdByPointers = new Map<Pointer, EntityId>()
+
+  const reportCacheAccess = (entity: Entity, result: 'hit' | 'miss') => {
+    components.metrics.increment('dcl_entities_cache_accesses_total', {
+      entity_type: entity.type,
+      result: result
+    })
+  }
+
+  const cacheEntity = (entity: Entity): void => {
+    cache.set(entity.id, entity)
+  }
+
+  const invalidateEntity = (entityId: EntityId): void => {
+    const entity = cache.get(entityId)
+    if (entity) {
+      for (const pointer of entity.pointers) {
+        entityIdByPointers.delete(pointer)
+      }
+    }
+    cache.del(entityId)
+  }
+
+  const updateCacheAndReturnAsEntities = (deployments: Deployment[]): Entity[] => {
+    const entities = mapDeploymentsToEntities(deployments)
+    // Save the calculated values
+    for (const entity of entities) {
+      cacheEntity(entity)
+      for (const pointer of entity.pointers) {
+        entityIdByPointers.set(pointer, entity.id)
+      }
+    }
+    return entities
+  }
+
+  /**
+   * Queries DB to retrieve deployments using the given ids as filter and return them as entities
+   * It also updates the cache
+   */
+  const findEntitiesByIds = async (entityIds: EntityId[]): Promise<Entity[]> => {
+    const { deployments } = await getDeployments(components, {
+      filters: { entityIds, onlyCurrentlyPointed: true }
+    })
+    return updateCacheAndReturnAsEntities(deployments)
+  }
+
+  /**
+   * Queries DB to retrieve deployments using the given pointers as filter and return them as entities
+   * It also updates the cache
+   */
+  const findEntitiesByPointers = async (pointers: Pointer[]): Promise<Entity[]> => {
+    const { deployments } = await getDeployments(components, {
+      filters: { pointers, onlyCurrentlyPointed: true }
+    })
+    for (const deployment of deployments) {
+      deployment.pointers.forEach((pointer) => entityIdByPointers.set(pointer, deployment.entityId))
+    }
+    return updateCacheAndReturnAsEntities(deployments)
+  }
+
+  const withIds = async (...entityIds: EntityId[]): Promise<Entity[]> => {
```

Comment:
Do you ever plan to call the function as in `withIds(a,b,c,d)` otherwise just call `entitiIds` the param